### PR TITLE
[ fix ] test suite following #5166

### LINKED ISSUE #5029: One can override --safe
Agda accepts the following code:
```agda
{-# OPTIONS --safe -WnoSafeFlagNoCoverageCheck #-}

open import Agda.Builtin.Bool
open import Agda.Builtin.Equality

data ⊥ : Set where

{-# NON_COVERING #-}

f : ∀ b → b ≡ true → ⊥
f false ()

bad : ⊥
bad = f true refl
```
I guess one way to fix this would be to raise an error if an attempt is made to turn off a non-benign warning when `--safe` is used.

### LINKED ISSUE #5169: Master fails test "Issue3436"
- MacOS 10.15.7, Xcode-12.4.
- GHC-8.10.3, Cabal-3.2.0.0.
- MacTeX-2020.

```
.  .  .  .  .
diff --git a/private/var/folders/_l/4q83bg9j5ysb7qd1n9xpnb4h0000gn/T/Issue343673746-0.golden b/private/var/folders/_l/4q83bg9j5ysb7qd1n9xpnb4h0000gn/T/Issue343673746-1.actual
index 699465fef..84757769d 100644
--- a/private/var/folders/_l/4q83bg9j5ysb7qd1n9xpnb4h0000gn/T/Issue343673746-0.golden
+++ b/private/var/folders/_l/4q83bg9j5ysb7qd1n9xpnb4h0000gn/T/Issue343673746-1.actual
@@ -1,2 +1,7 @@
-AGDA_SUCCESS
+AGDA_UNEXPECTED_FAIL
 
+ret > ExitFailure 42
+out > Issue3436.agda:1,1-39
+out > Warnings for non-fatal errors such as MissingDefinitions cannot be
+out > turned off. See --help=warning.
+out >

Issue3436:                                                    Golden value differs from actual value
.  .  .  .  .
1 out of 1576 tests failed (156.45s)
```

### PR DESCRIPTION
**Pull request by gallais**

### CODE REVIEW

**Comment by gallais** [comment]
Travis did not run so I have no clue whether this will work
(apart from running `make test` locally).

**Comment by L-TChen** [comment]
You should check https://github.com/gallais/agda/actions?query=workflow%3A%22Build%2C+Test%2C+and+Benchmark%22 to see if it passes all checks on your fork.

File: src/full/Agda/Interaction/Options/Warnings.hs [hunk 1]
```diff
@@ -58,19 +60,37 @@ defaultWarningMode :: WarningMode
 defaultWarningMode = WarningMode ws False where
   ws = fst $ fromMaybe __IMPOSSIBLE__ $ lookup defaultWarningSet warningSets
 
+-- | Some warnings are errors and cannot be turned off.
+data WarningModeError = Unknown String | NoNoError String
+
+prettyWarningModeError :: WarningModeError -> String
+prettyWarningModeError = \case
+  Unknown str -> concat [ "Unknown warning flag: ", str, "." ]
+  NoNoError str -> concat [ "Warnings for non-fatal errors such as "
```
Comment (andreasabel) [inline src/full/Agda/Interaction/Options/Warnings.hs:69]: This wording sounds like if warnings for _fatal errors_ __could__ be turned off, which is probably not the intended connotation.

**Review by andreasabel — COMMENTED** [review]
Clarify wording of error message, please.

**Pull request merged**